perun/collect/trace/locks.py

```diff
@@ -0,0 +1,231 @@
+""" The 'locks' module provides the necessary features for locking some of the resources
+used by the trace collector. Namely the resources contained within the LockType enum.
+
+The locks are needed to ensure that certain resources are not re-used by another running
+perun process which could introduce output data corruption, e.g. when the same binary file
+is used by two instances of trace collector, the output data could get mixed in both of the
+output files - the same would happen if one SystemTap kernel module is used by two different
+SystemTap instances etc.
+
+The locks are represented by an object which can - upon the locking operation - create a
+corresponding lock file in the .perun directory. The locking should not be perceived as it is
+typically (semaphores, monitors or other locking primitives) used in a multiprocess /
+multithread programs - here we are dealing with potentially multiple unrelated and not
+connected perun processes that may be attempting to use the same resource. Thus the files
+are used as a mean to communicate with other running trace collectors that certain resources
+should not be used.
+
+However, since the locks are stored in a .perun directory, the scope of the locks is limited
+to the corresponding perun instance and does not cover the whole system, i.e. two perun
+processes originating from different perun instances will not be able to successfully lock
+their resources against each other.
+"""
+
+import os
+from enum import Enum
+
+from perun.collect.trace.watchdog import WD
```

Comment:
Suggestion: Rename to something longer. WATCH_DOG?

Reply:
Changed.